### PR TITLE
Implement armstrong numbers

### PR DESCRIPTION
**Pull request by marvelou-s**
Hi!

Ported [armstrong-numbers](https://github.com/exercism/problem-specifications/tree/master/exercises/armstrong-numbers) exercise to C++.

### CODE REVIEW

File: exercises/armstrong-numbers/example.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef ARMSTRONG_NUMBERS_H_
+#define ARMSTRONG_NUMBERS_H_
+
+namespace armstrong_numbers {
+
+bool is_armstrong_number(int number);
+
+} // namespace armstrong_numbers
+
+#endif //!ARMSTRONG_NUMBERS_
```
Comment (patricksjackson) [inline exercises/armstrong-numbers/example.h]: Nit: this doesn't match the above `#ifndef`.

File: exercises/armstrong-numbers/example.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#ifndef ARMSTRONG_NUMBERS_H_
```
Comment (patricksjackson) [inline exercises/armstrong-numbers/example.h]: Nit: All the other exercises AFAIK use `#if !defined(ARMSTRONG_NUMBERS_H_)`

File: exercises/armstrong-numbers/armstrong_numbers_test.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#if defined(EXERCISM_RUN_ALL_TESTS)
+
+TEST_CASE("single_digit_numbers_are_armstrong_numbers")
+{
+    REQUIRE(armstrong_numbers::is_armstrong_number(5));
+}
+
+TEST_CASE("there_are_no_2_digit_armstrong_numbers")
+{
+    REQUIRE(!armstrong_numbers::is_armstrong_number(10));
```
Comment (patricksjackson) [inline exercises/armstrong-numbers/armstrong_numbers_test.cpp]: I just realized, Catch has a `REQUIRE_FALSE` macro that would probably make these test cases clearer even though we haven't used it in other test suites. Maybe it's worth it to use that macro instead for clarity?

File: exercises/armstrong-numbers/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Get the exercise name from the current directory
```
Comment (patricksjackson) [inline exercises/armstrong-numbers/CMakeLists.txt]: I made a change to the default CMakeLists.txt file after you made this. See #276. Can you update this file?

**Comment by marvelou-s** [comment]
@patricksjackson thank you for your time! I like your proposal for make example solutions as clear as we can for future students. Note for self for my next PRs ;)

**Pull request merged**